src/rpdk/core/contract/resource_client.py

```diff
@@ -69,6 +69,24 @@ def prune_properties_if_not_exist_in_path(output_model, input_model, paths):
     return output_document["properties"]
 
 
+def prune_properties_which_dont_exist_in_path(model, paths):
+    """Prune given properties from a model. This method removes any property
+    from the model which does not exists in the paths
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function returns the model after pruning the path which exists
+    in the paths tuple but not in the input_model
```

Comment:
method description needs improvement. I think it is pruning properties which exist in path and returning those.

Reply:
update the description